server/game/player.js

```diff
@@ -391,6 +391,7 @@ class Player extends Spectator {
 
         if(card.getType() === 'attachment' && this.phase !== 'setup' && !dupeCard) {
             this.promptForAttachment(card);
+            this.game.raiseEvent('onCardEntersPlay', card);
```

Comment:
Good catch! Though unless @cryogen has other ideas I think adding this to the `Player.attach()` method instead would be more appropriate. The attachment doesn't actually enter play here, the player is only prompted to attach it to something. The player can cancel that prompt, so the card isn't guaranteed to enter play.